vf_convenience/cl_menu.lua

```diff
@@ -0,0 +1,109 @@
+CurrentSlot = 10
+gxt = "SNK_MNU"
+
+
+RegisterNetEvent('vf_convenience:update')
+AddEventHandler('vf_convenience:update', function()
+  if DoesStoreMenuExist then
+    _generalStorePool:Clear()
+    _generalStorePool:Remove()
+
+    -- Create the store menu
+    _generalStorePool = NativeUI.CreatePool()
+    storeMenu = NativeUI.CreateMenu("", "~s~" .. GetLabelText("SNK_ITEM"), "", "", MenuType, MenuType)
+    _generalStorePool:Add(storeMenu)
+
+    _generalStorePool:ControlDisablingEnabled(true)
+    _generalStorePool:MouseControlsEnabled(false)
+
+    Wait(10)
+
+    CreateStoreMenu(storeMenu)
+    _generalStorePool:RefreshIndex()
+    storeMenu:Visible(not storeMenu:Visible())
+    DoesStoreMenuExist = true
+  end
+end)
+
+Citizen.CreateThread(function()
+  while not NetworkIsGameInProgress() and not IsPlayerPlaying(PlayerId()) do
+    Citizen.Wait(0)
+  end
+
+  while HasAdditionalTextLoaded(CurrentSlot) and not HasThisAdditionalTextLoaded(gxt, CurrentSlot) do
+    Citizen.Wait(0)
+    CurrentSlot = CurrentSlot + 1
+  end
+
+  if not HasThisAdditionalTextLoaded(gxt, CurrentSlot) then
+    ClearAdditionalText(CurrentSlot, true)
+    RequestAdditionalText(gxt, CurrentSlot)
+    while not HasThisAdditionalTextLoaded(gxt, CurrentSlot) do
+      Citizen.Wait(0)
+    end
+  end
+
+  Citizen.Wait(1800)
+end)
+
+function CreateStoreMenu(menu)  
+  inventoryItems = exports.vf_base:GetInventory()
```

Comment:
It's a good practice to define variables as local as long as you don't need them outside that scope so they don't leak over accidentally.